p2p/libp2p/message.go

```diff
@@ -31,11 +31,12 @@ func NewMessage(message *pubsub.Message) *Message {
 	id, err := peer.IDFromBytes(msg.from)
 	if err != nil {
 		log.Trace("creating new p2p message", "error", err.Error())
+		return nil, err
 	} else {
```

Comment:
You can drop the whole else now, and add in line 38 this line: msg.peer = p2p.PeerID(id)

Reply:
done